pkg/controllers/streaming/processor_controller_test.go

```diff
@@ -0,0 +1,558 @@
+/*
+Copyright 2019 the original author or authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package streaming
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/go-logr/logr"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/projectriff/system/pkg/apis"
+	"github.com/projectriff/system/pkg/apis/build/v1alpha1"
+	corev1alpha1 "github.com/projectriff/system/pkg/apis/core/v1alpha1"
+	streamingv1alpha1 "github.com/projectriff/system/pkg/apis/streaming/v1alpha1"
+	kedav1alpha1 "github.com/projectriff/system/pkg/apis/thirdparty/keda/v1alpha1"
+	rtesting "github.com/projectriff/system/pkg/controllers/testing"
+	"github.com/projectriff/system/pkg/tracker"
+)
+
+const (
+	testNamespace      = "test-namespace"
+	testName           = "test-processor"
+	testProcessorImage = "test-processor-image"
+	testDefaultImage   = "test-image-from-template"
+	testFunction       = "test-function"
+	testFunctionImage  = "test-function-image"
+	testContainer      = "test-container"
+	testContainerImage = "test-container-image"
+	testErrorMessage   = "test error"
+)
+
+var testError = errors.New(testErrorMessage)
+
+func TestReconcile(t *testing.T) {
+	table := rtesting.Table{{
+		Name:        "processor does not exist",
+		Key:         rtesting.Key(testNamespace, testName),
+		ExpectTrack: []rtesting.TrackRequest{},
+	}, {
+		Name:             "getting processor fails",
+		Key:              rtesting.Key(testNamespace, testName),
+		ExpectTrack:      []rtesting.TrackRequest{},
+		GetHook:          getErrorOn(testNamespace, testName),
```

Comment:
for what resource Kind? We will frequently have different types of resources with the same name in the same namespaces that are related.

Reply:
Not sure I follow. `Kind` isn't part of the signature of `Get`.

Reply:
I am reminded that `Kind` can be inferred from the concrete type of the object pointer passed to `Get` by using the scheme.

Reply:
Fixed.